controllers/cluster_controller.go

```diff
@@ -45,24 +39,19 @@ import (
 	"github.com/go-logr/logr"
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/tools/record"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var instanceIDRegex = regexp.MustCompile(`^([\w-]+)-(\d+)`)
-var processIDRegex = regexp.MustCompile(`^([\w-]+-\d)-\d$`)
-
 // FoundationDBClusterReconciler reconciles a FoundationDBCluster object
 type FoundationDBClusterReconciler struct {
 	client.Client
 	Recorder            record.EventRecorder
 	Log                 logr.Logger
 	InSimulation        bool
 	PodLifecycleManager PodLifecycleManager
-	PodClientProvider   func(*fdbtypes.FoundationDBCluster, *corev1.Pod) (FdbPodClient, error)
+	PodClientProvider   func(*fdbtypes.FoundationDBCluster, *corev1.Pod) (internal.FdbPodClient, error)
```

Comment:
If this field refers to an internal type, does it mean that something that imports the `controllers` package would be unable to create a `FoundationDBClusterReconciler`?

Reply:
Oh right, I'll check that. Good point 👍

Reply:
I'll created a method for that, which only exposes fields that should be changed. We could also think about exposing a smaller surface before 1.0 but I would solve that in another PR.